packages/manager/src/features/IAM/Shared/Permissions/Permissions.tsx

```diff
@@ -0,0 +1,105 @@
+import { StyledLinkButton, TooltipIcon } from '@linode/ui';
+import { debounce } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import * as React from 'react';
+
+import {
+  StyledBox,
+  StyledClampedContent,
+  StyledContainer,
+  StyledGrid,
+  StyledListItem,
+  StyledSpan,
+  StyledTypography,
+  sxTooltipIcon,
+} from './Permissions.style';
+
+import type { PermissionType } from '@linode/api-v4/lib/iam/types';
+
+type Props = {
+  permissions: PermissionType[];
+};
+
+export const Permissions = ({ permissions }: Props) => {
+  const [showAll, setShowAll] = React.useState(false);
+  const [numHiddenItems, setNumHiddenItems] = React.useState<number>(0);
+
+  const containerRef = React.useRef<HTMLDivElement | null>(null);
+
+  const calculateHiddenItems = React.useCallback(() => {
+    if (showAll) {
+      setNumHiddenItems(0);
+      return;
+    }
+
+    const containerRect = containerRef.current?.getBoundingClientRect();
+    if (!containerRect) {
+      return;
+    }
+
+    const items = containerRef.current?.querySelectorAll('p');
+    if (!items) {
+      return;
+    }
+
+    const containerBottom = containerRect.top + containerRect.height;
+
+    const itemsArray = Array.from(items);
+    const firstHiddenIndex = itemsArray.findIndex(
+      (item: HTMLParagraphElement) => {
+        const rect = item.getBoundingClientRect();
+        return rect.top >= containerBottom;
+      }
+    );
+
+    const numHiddenItems = firstHiddenIndex
+      ? itemsArray.length - firstHiddenIndex
+      : 0;
+
+    setNumHiddenItems(numHiddenItems);
```

Comment:
With the `ref` approach in the below (or above?) comment, we can now clean this up to achieve the same functionality. 👍 
```suggestion
    let hiddenCount = 0;

    for (const item of itemRefs.current) {
      if (item) {
        const itemRect = item.getBoundingClientRect();
        const itemTopRelative = itemRect.top - containerRect.top;

        if (itemTopRelative >= containerRect.height) {
          hiddenCount++;
        }
      }
    }

    setNumHiddenItems(hiddenCount);
```

Reply:
I wouldn't change the find for a loop. There is no need to iterate through the entire list once you've found one that is below the fold.

Reply:
That's a fair point 👍